arcsi/api/item.py

```diff
@@ -422,17 +482,234 @@ def archon_edit_item(id):
+                    play_url_name = item.play_file_name
+            else:
+                (tmp_file_saved, play_url_name) = do.tmp_save_file(
+                    item.shows[0].archive_lahmastore_base_url,
+                    item.archive_lahmastore_canonical_url,
+                    item.number,
+                )
+                temp_urls.append(tmp_file_saved)
+            if request_files["image_file"] and request_files.get("image_file"):
+                if not item.image_url:
+                    image_url_name = save_file(
+                        archive_base=item.shows[0].archive_lahmastore_base_url,
+                        archive_idx=item.number,
+                        archive_file=request_files["image_file"],
+                        archive_file_name=(item.shows[0].name, item_name),
+                    )
+                    if not image_url_name:
+                        temp_urls.append(False)
+                else:
+                    image_url_name = item.image_url.rsplit("/", 1)[1]
+            else:
+                if item.shows[0].cover_image_url:
+                    fallback_image = item.shows[0].cover_image_url
+                else:
+                    fallback_image = "/static/img/lahmacun-logo.png"
+
+                (tmp_image_saved, image_url_name) = do.tmp_save_file(
+                    item.shows[0].archive_lahmastore_base_url,
+                    fallback_image,
+                    0,
+                )
+                temp_urls.append(tmp_image_saved)
+            if temp_urls:
+                for saved in temp_urls:
+                    if not saved:
+                        return make_response(
+                            jsonify(
+                                {
+                                    "error": {
+                                        "message": "Episode not found in storage. The system provided the following error.",
+                                        "errorReason": "ERROR: Request to {} returned {}: {},".format(
+                                            presigned, resp.status_code, resp.reason
+                                        ),
+                                        "code": 10201030,
+                                    }
+                                },
+                                404,
+                                headers,
+                            )
+                        )
+
+            item.airing = broadcast_audio(
+                archive_base=item.shows[0].archive_lahmastore_base_url,
+                archive_idx=item.number,
+                broadcast_file_name=play_url_name,
+                broadcast_playlist=item.shows[0].playlist_name,
+                broadcast_show=item.shows[0].name,
+                broadcast_title=item.name,
+                image_file_name=image_url_name,
+            )
+
+            if item.airing:
+                db.session.commit()
+            # TODO cleanup tmp files after upload?
```

Comment:
Nice idea, this will help out the daily tmpreaper job, and it will also decrease the chance that the show hosts receive a storage related error.
I added this into the other PR in case of successful uploads!

Reply:
Great!